### ISSUE TITLE
Hidden property doesn't work when the type of the field is a custom type of type array

### ISSUE THREAD

**Issue by fvieira** [sanity-io/sanity#1864]
**Describe the bug**

See below, the field whose type is directly an array is properly hidden, but the field whose type is arrayOfStrings which is itself an array doesn't get hidden. Both fields work the same otherwise, so I see no reason for the hidden property to work differently.

``` javascript
{
    name: 'arrayOfStrings',
    type: 'array',
    of: [{ type: 'string' }],
  },
  {
    name: 'myDoc',
    title: 'My Doc',
    type: 'document',
    fields: [
      {
        name: 'myVisibleArrayOfStrings',
        title: 'My Visible Array of Strings',
        type: 'arrayOfStrings',
        hidden: true,
      },
      {
        name: 'myHiddenArrayOfStrings',
        title: 'My Hidden Array of Strings ',
        type: 'array',
        of: [{ type: 'string' }],
        hidden: true,
      },
    ],
  },
```

**My use case:**

I'm building a cms for a multinational where each country will have it's own instance of the cms. The structure is mostly the same for all countries but for some exceptions, so I want to use the same codebase for all but hide some of the fields on some of the countries. Also, I have extracted all portable text types (arrays of blocks and other objects) into a separate file so that I can reuse them across the whole cms. This means that these portable text types (being of type array) cannot be hidden due to this bug.

**Which versions of Sanity are you using?**

@sanity/cli                 1.149.9 (up to date)
@sanity/base               1.149.10 (up to date)
@sanity/cli                 1.149.9 (up to date)
@sanity/components         1.149.10 (up to date)
@sanity/core               1.149.12 (up to date)
@sanity/dashboard          1.149.10 (up to date)
@sanity/default-layout     1.149.10 (up to date)
@sanity/default-login      1.149.11 (up to date)
@sanity/desk-tool          1.149.10 (up to date)
@sanity/google-maps-input   1.149.7 (up to date)
@sanity/storybook           1.149.7 (up to date)
@sanity/vision              1.149.0 (up to date)

**Let me know if you fail to reproduce and I'll provide more details then.**

**Comment by kmelve** [comment]
This indeed seems like a bug, thanks for reporting!

**Comment by masives** [comment]
I believe it's not only issue with array types but also inside an array.  In example below the image element is hidden but left out for backwars compatibility. However, it still shows up in studio
```js
{
  title: 'Block Content',
  name: 'blockContent',
  type: 'array',
  of: [
    {
      title: 'Block',
      type: 'block',
      styles: [
        { title: 'Normal', value: 'normal' },
        { title: 'H1', value: 'h1' },
        { title: 'H2', value: 'h2' },
        { title: 'H3', value: 'h3' },
        { title: 'H4', value: 'h4' },
        { title: 'Quote', value: 'blockquote' },
      ],
      lists: [{ title: 'Bullet', value: 'bullet' }, { title: 'Number', value: 'number' }],
      marks: {
        decorators: [{ title: 'Strong', value: 'strong' }, { title: 'Emphasis', value: 'em' }],
        annotations: [
          {
            title: 'URL',
            name: 'link',
            type: 'object',
            fields: [
              {
                title: 'URL',
                name: 'href',
                type: 'url',
              },
            ],
          },
          {
            title: 'Text accent color',
            name: 'color',
            type: 'color',
          },
        ],
      },
    },
// HERE IS THE IMAGE TYPE THAT DOESN'T GET HIDDEN
    {
      type: 'image',
      description: 'Basic image for backwards compatibility',
      options: { hotspot: true },
      hidden: true
    },
    {
      title: 'Enhanced Image',
      type: 'enhancedImage',
      description: 'enhanced image with caption and alt',
      options: { hotspot: true },
    },
    {
      title: 'Code',
      description: 'Code editor',
      type: 'code',
    },
  ],
};

```

**Comment by vicbergquist** [comment]
Thanks for reporting! Found the problem and made a PR to fix it 😊 https://github.com/sanity-io/sanity/pull/1946

**Comment by vicbergquist** [comment]
> I believe it's not only issue with array types but also inside an array. In example below the image element is hidden but left out for backwars compatibility. However, it still shows up in studio
> 
> ```js
> {
>   title: 'Block Content',
>   name: 'blockContent',
>   type: 'array',
>   of: [
>     {
>       title: 'Block',
>       type: 'block',
>       styles: [
>         { title: 'Normal', value: 'normal' },
>         { title: 'H1', value: 'h1' },
>         { title: 'H2', value: 'h2' },
>         { title: 'H3', value: 'h3' },
>         { title: 'H4', value: 'h4' },
>         { title: 'Quote', value: 'blockquote' },
>       ],
>       lists: [{ title: 'Bullet', value: 'bullet' }, { title: 'Number', value: 'number' }],
>       marks: {
>         decorators: [{ title: 'Strong', value: 'strong' }, { title: 'Emphasis', value: 'em' }],
>         annotations: [
>           {
>             title: 'URL',
>             name: 'link',
>             type: 'object',
>             fields: [
>               {
>                 title: 'URL',
>                 name: 'href',
>                 type: 'url',
>               },
>             ],
>           },
>           {
>             title: 'Text accent color',
>             name: 'color',
>             type: 'color',
>           },
>         ],
>       },
>     },
> // HERE IS THE IMAGE TYPE THAT DOESN'T GET HIDDEN
>     {
>       type: 'image',
>       description: 'Basic image for backwards compatibility',
>       options: { hotspot: true },
>       hidden: true
>     },
>     {
>       title: 'Enhanced Image',
>       type: 'enhancedImage',
>       description: 'enhanced image with caption and alt',
>       options: { hotspot: true },
>     },
>     {
>       title: 'Code',
>       description: 'Code editor',
>       type: 'code',
>     },
>   ],
> };
> ```

This is actually a different bug than the original one, but the fix will hopefully be out soon 👍

**Comment by fvieira** [comment]
> Thanks for reporting! Found the problem and made a PR to fix it 😊 #1946

That's great news! Thank you! 😁
